### PR TITLE
[REF] Reconcile CRM_Utils_System::getUrlPath and CRM_Utils_System::currentPath

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
Code refactoring to make looking up the current URL more consistent.

Before
----------------------------------------
- 2 functions that did the same thing (retrieve current url).
- Lots of code using neither function but doing it the hard way directly accessing `$_GET`.

After
----------------------------------------
- Deprecated `CRM_Utils_System::getUrlPath`.
- Switched code to use the non-deprecated `CRM_Utils_System::currentPath`.
- Refactored code accessing `$_GET` directly to use the helper function.

### CODE REVIEW

File: CRM/Report/Utils/Report.php [hunk 1]
```diff
@@ -298,31 +297,22 @@ public static function makeCsv(&$form, &$rows) {
...
-    $arg = explode('/', $_GET[$config->userFrameworkURLVar]);
+    $arg = explode('/', CRM_Utils_System::currentPath());
 
-    if ($arg[1] == 'report' &&
-      CRM_Utils_Array::value(2, $arg) == 'instance'
-    ) {
-      if (CRM_Utils_Rule::positiveInteger($arg[3])) {
-        return $arg[3];
-      }
+    if (isset($arg[3]) && $arg[1] == 'report' && $arg[2] == 'instance' && CRM_Utils_Rule::positiveInteger($arg[3])) {
```
Comment (seamuslee001) [inline CRM/Report/Utils/Report.php:302]: @colemanw what if there is no `2` key? I guess that would be covered by the first isset right?

Comment (colemanw) [reply CRM/Report/Utils/Report.php:302]: Right. It's always going to be a non-associative array, so if `[3]` is set then so are `[2]` and `[1]` and `[0]`.

**Comment by kcristiano** [comment]
@colemanw I'll do some `r-run` on the various configurations that caused us url and path trouble in 5.23 on Joomla and WP.  Will likely be next week.

**Comment by mattwire** [comment]
@colemanw What is not obvious from the description is which one is deprecated! It *is* obvious looking at the code but would help reviewers :-)

**Comment by colemanw** [comment]
Thanks @kcristiano for testing.
@mattwire I've updated the description.

**Comment by kcristiano** [comment]
@colemanw I've done `r-run` on WP and Joomla and have not hit any errors.

I think this should be merged so we can have in the next RC

**Pull request merged**